zegami_sdk/collection.py

```diff
@@ -408,6 +520,17 @@ def get_image_urls(self, rows=None, source=0, generate_signed_urls=False,
 
             return signed_route_urls
 
+    def get_feature_extraction_imageset_id(self, source=0) -> str:
+        """Returns the feature extraction imageset id in the given source index."""
+        source = self._parse_source(source)
+        source_name = source.name
+        all_nodes = self.node_statuses
+        for node in all_nodes:
+            if (re.search("^Feature extraction imageset", node['name'])
```

Comment:
Not really safe to search by name. Better to filter by the node's action and source node_group. No guarantees there will only be one, but could just return the first one found (or None).
Probably out of scope for this piece of work, but maybe `add_feature_pipeline` should support adding a new feature extraction node that uses a different model.